crates/hyperswitch_connectors/src/connectors/fiservemea/transformers.rs

```diff
@@ -56,66 +92,261 @@ impl TryFrom<&FiservemeaRouterData<&PaymentsAuthorizeRouterData>> for Fiservemea
     ) -> Result<Self, Self::Error> {
         match item.router_data.request.payment_method_data.clone() {
             PaymentMethodData::Card(req_card) => {
-                let card = FiservemeaCard {
+                let card = FiservemeaPaymentCard {
                     number: req_card.card_number,
-                    expiry_month: req_card.card_exp_month,
-                    expiry_year: req_card.card_exp_year,
-                    cvc: req_card.card_cvc,
-                    complete: item.router_data.request.is_auto_capture()?,
+                    expiry_date: FiservemeaExpiryDate {
+                        month: req_card.card_exp_month,
+                        year: req_card.card_exp_year,
+                    },
+                    security_code: req_card.card_cvc,
+                };
+                let request_type = if item.router_data.request.capture_method
+                    == Some(enums::CaptureMethod::Automatic)
+                {
+                    FiservemeaRequestType::PaymentCardSaleTransaction
+                } else {
+                    FiservemeaRequestType::PaymentCardPreAuthTransaction
                 };
                 Ok(Self {
-                    amount: item.amount.clone(),
-                    card,
+                    request_type,
+                    merchant_transaction_id: item
+                        .router_data
+                        .connector_request_reference_id
+                        .clone(),
+                    transaction_amount: FiservemeaTransactionAmount {
+                        total: item.amount.clone(),
+                        currency: item.router_data.request.currency,
+                    },
+                    order: FiservemeaOrder {
+                        order_id: item.router_data.connector_request_reference_id.clone(),
+                    },
```

Comment:
Either `merchant_transaction_id` or `order` we should pass merchant_order_reference_id, if its available else pass `connector_request_reference_id`

Reply:
Done